app/models/teacher_set.rb

```diff
@@ -102,6 +102,41 @@ def holds_for_user(user, hold_id)
     end
   end
 
+  
+  def teacher_set_availability
+    self.available_copies.to_i > 0 ? AVAILABLE : UNAVAILABLE
+  end
+
+
+  # calculate teacher-set available copies while creating/cancelling the hold
+  def calculate_available_copies(status, quantity, current_user=nil, hold_id=nil)
+    if status == 'cancelled'
+      user_holds_count = holds_count_for_user(current_user, hold_id).to_i
+      available_copies = self.available_copies.to_i + user_holds_count
+    else
+      available_copies = self.available_copies - quantity.to_i
+    end
+    available_copies
+  end
+
+
+  def update_teacher_set_availability_in_db(status, quantity=nil, current_user=nil, hold_id=nil)
+    available_copies, availability = calculate_available_copies(status, quantity, current_user, hold_id)
+    self.available_copies = available_copies
+    self.availability = teacher_set_availability
```

Comment:
Based on your last comment at planning — about the need to think through implications of changing `availability_count` etc., should `availability` be saved to the DB? Could we just remove the column from the TeacherSet table, and rename the `teacher_set_availability` method to `availability`? Happy to discuss further — let me know what you think!

Reply:
That way there would be no way for the `availability` column and `available_copies` to be inconsistent.